app/services/doc_auth/lexis_nexis/responses/true_id_response.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module DocAuth
+  module LexisNexis
+    module Responses
+      class TrueIdResponse < LexisNexisResponse
+        def initialize(http_response)
+          super http_response
+        end
+
+        def successful_result?
+          transaction_status == 'passed' &&
+            product_status == 'pass' &&
+            doc_auth_result == 'Passed'
```

Comment:
this is soooo weird to me that there are all these different spellings & casings of `pass` vs `passed`, WDYT if we pulled these strings into each method so that they stay closer the keys/values they are sourced from? ex:

```ruby
def successful_result?
  transaction_status_passed? &&
    product_status_passed? &&
    doc_auth_result_passed?
end

def transaction_status_passed?
  status.dig(:TransactionStatus) == 'passed'
end
```

(I also removed the memoization, because for true/false values it's not useful, and dictionary lookups are cheap. IMO memoization is best for expensive values like allocations or DB queries)

Reply:
yes, it is very awkward. good call.